### DISCUSSION TITLE
Follow external URLS

### DISCUSSION THREAD

**Discussion by Liquidmantis** [nvim-telekasten/telekasten.nvim#26] [Ideas]
One of the things I immediately see when fully replacing vimwiki with telekasten is that I'm not able to follow external URLs, e.g `[Use me to search!](https://google.com)`.  Trying to follow external links throws an error:

```
E5108: Error executing lua ...ite/pack/packer/start/telekasten.nvim/lua/telekasten.lua:244: attempt to index local 'ofile' (a nil value)   stack traceback:                                                                                                                                   ...ite/pack/packer/start/telekasten.nvim/lua/telekasten.lua:244: in function 'create_note_from_template'                           
        ...ite/pack/packer/start/telekasten.nvim/lua/telekasten.lua:742: in function 'follow_link'                                         
        [string ":lua"]:1: in main chunk                                               
```

(let me know if these should be issues rather than discussion threads.  It seems like all the activity is over here.)

**Comment by lambtho12** [comment]
This is more a general (n)vim mapping. Usually you just have to do `gx`to follow a URL.

For some reason it does not work on my config (and I am too lazy to find out why), so I have the following explicitely mapped:
```lua
if vim.fn.has("mac") == 1 then
  map(
    "n",
    "gx",
    '<Cmd>call jobstart(["open", expand("<cfile>")], {"detach": v:true})<CR>'
  )
elseif vim.fn.has("unix") == 1 then
  map(
    "n",
    "gx",
    '<Cmd>call jobstart(["xdg-open", expand("<cfile>")], {"detach": v:true})<CR>'
  )
else
  map("n", "gx", '<Cmd>lua print("Error: gx is not supported on this OS!")<CR>')
end

```

[source](https://sbulav.github.io/vim/neovim-opening-urls/)

**Comment by renerocksai** [comment]
Awesome! I know about `open` (mac) and `xdg-open` (Linux), so this is a cool way to open external links. I didn't think of adding this but it makes total sense! Probably more for external links that don't seem to go to the filesystem but have a `{protocol}://`, like `https://` or `some-app-integration://`.

BTW: that error message just shouldn't happen. If a file doesn't exist we should not even try to open it :smile:

**Comment by renerocksai** [comment]
Posting here is fine. As long as it's marked as idea, it's all easy to find :smile:

**Comment by renerocksai** [comment]
Done! :slightly_smiling_face:

**Reply by Liquidmantis** [reply]
Whoa!  I haven't even had time to get updated on all the new changes since yesterday!
